log/logClient.go

```diff
@@ -192,8 +193,25 @@ func (fd *Feeder) WatchMessages(msgPath string, jsonFormat bool) error {
 	return nil
 }
 
+func RegexMatcher(filter string, res string, caseSensitive bool) (bool, error) {
+
+	if caseSensitive {
+		filter = "(?i)" + filter
+	}
+
+	re, err := regexp.Compile(filter)
+	if err != nil {
+		return false, err
+	}
```

Comment:
We are still compiling regex on every alert/log. Ideally we should compile the regex once, so even before the options are passed to WatchAlerts/WatchLogs.